pkg/resources/resource_source_load_generator.go

```diff
@@ -119,28 +165,67 @@ func sourceLoadgenCreate(ctx context.Context, d *schema.ResourceData, meta any)
 		builder.LoadGeneratorType(v.(string))
 	}
 
-	if v, ok := d.GetOk("tick_interval"); ok {
-		builder.TickInterval(v.(string))
-	}
+	if v, ok := d.GetOk("counter_options"); ok {
```

Comment:
This is ugly code. Trying to think of a more elegant way to coalesce the values while still being safe.